web3_proxy/src/app/mod.rs

```diff
@@ -608,6 +608,104 @@ impl Web3ProxyApp {
                 .build()?,
         );
 
+        // // TODO: i don't like doing Block::default here! Change this to "None"?
+        // let (watch_consensus_head_sender, watch_consensus_head_receiver) = watch::channel(None);
```

Comment:
yeah delete both. lines

Reply:
@WyseNynja I was referring to these lines here in app/mod.rs:

```
// // TODO: i don't like doing Block::default here! Change this to "None"?
        // let (watch_consensus_head_sender, watch_consensus_head_receiver) = watch::channel(None);
        // // TODO: will one receiver lagging be okay? how big should this be?
        // let (pending_tx_sender, pending_tx_receiver) = broadcast::channel(256);
        //
        // // TODO: use this? it could listen for confirmed transactions and then clear pending_transactions, but the head_block_sender is doing that
        // // TODO: don't drop the pending_tx_receiver. instead, read it to mark transactions as "seen". once seen, we won't re-send them?
        // // TODO: once a transaction is "Confirmed" we remove it from the map. this should prevent major memory leaks.
        // // TODO: we should still have some sort of expiration or maximum size limit for the map
        // drop(pending_tx_receiver);
        //
        // // TODO: capacity from configs
        // // all these are the same size, so no need for a weigher
        // // TODO: ttl on this? or is max_capacity fine?
        // let pending_transactions = Cache::builder()
        //     .max_capacity(10_000)
        //     // TODO: different chains might handle this differently
        //     // TODO: what should we set? 5 minutes is arbitrary. the nodes themselves hold onto transactions for much longer
        //     .time_to_idle(Duration::from_secs(300))
        //     .build_with_hasher(hashbrown::hash_map::DefaultHashBuilder::default());
        //
        // // keep 1GB/5 minutes of blocks in the cache
        // // TODO: limits from config
        // // these blocks don't have full transactions, but they do have rather variable amounts of transaction hashes
        // // TODO: how can we do the weigher better?
        // let block_map: BlockHashesCache = Cache::builder()
        //     .max_capacity(1024 * 1024 * 1024)
        //     .weigher(|_k, v: &Web3ProxyBlock| {
        //         // TODO: is this good enough?
        //         1 + v.block.transactions.len().try_into().unwrap_or(u32::MAX)
        //     })
        //     // TODO: what should we set? 5 minutes is arbitrary. the nodes themselves hold onto transactions for much longer
        //     .time_to_idle(Duration::from_secs(300))
        //     .build_with_hasher(hashbrown::hash_map::DefaultHashBuilder::default());
        //
        // // connect to the load balanced rpcs
        // let (balanced_rpcs, balanced_handle, consensus_connections_watcher) = Web3Rpcs::spawn(
        //     block_map.clone(),
        //     top_config.app.chain_id,
        //     db_conn.clone(),
        //     http_client.clone(),
        //     top_config.app.max_block_age,
        //     top_config.app.max_block_lag,
        //     top_config.app.min_synced_rpcs,
        //     top_config.app.min_sum_soft_limit,
        //     pending_transactions.clone(),
        //     Some(pending_tx_sender.clone()),
        //     vredis_pool.clone(),
        //     balanced_rpcs,
        //     Some(watch_consensus_head_sender),
        // )
        // .await
        // .context("spawning balanced rpcs")?;
        //
        // // save the handle to catch any errors
        // cancellable_handles.push(balanced_handle);
        //
        // // connect to the private rpcs
        // // only some chains have this, so this is optional
        // let private_rpcs = if private_rpcs.is_empty() {
        //     // TODO: do None instead of clone?
        //     warn!("No private relays configured. Any transactions will be broadcast to the public mempool!");
        //     None
        // } else {
        //     let (private_rpcs, private_handle, _) = Web3Rpcs::spawn(
        //         block_map,
        //         top_config.app.chain_id,
        //         db_conn.clone(),
        //         http_client.clone(),
        //         // private rpcs don't get subscriptions, so no need for max_block_age or max_block_lag
        //         None,
        //         None,
        //         0,
        //         0,
        //         pending_transactions.clone(),
        //         // TODO: subscribe to pending transactions on the private rpcs? they seem to have low rate limits, but they should have
        //         None,
        //         vredis_pool.clone(),
        //         private_rpcs,
        //         // subscribing to new heads here won't work well. if they are fast, they might be ahead of balanced_rpcs
        //         // they also often have low rate limits
        //         // however, they are well connected to miners/validators. so maybe using them as a safety check would be good
        //         // TODO: but maybe we could include privates in the "backup" tier
        //         None,
        //     )
        //     .await
        //     .context("spawning private_rpcs")?;
        //
        //     if private_rpcs.by_name.is_empty() {
        //         None
        //     } else {
        //         // save the handle to catch any errors
        //         cancellable_handles.push(private_handle);
        //
        //         Some(private_rpcs)
        //     }
        // };
    ```
    
    I am removing these comments for now, it seems repeated below (uncommented)